gb-cpu/src/memory.rs

```diff
@@ -0,0 +1,103 @@
+pub mod area;
+mod consts;
+
+use crate::error::Error;
+
+use crate::bus::Bus;
+use area::rom::{Mbc, NoMbc};
+use area::{Area, Wram};
+
+type Rom = Box<dyn Bus<usize, Item = u8, Result = Result<(), Error>, Data = u8>>;
+
+pub struct Memory {
+    pub wram: Wram,
+    pub rom: Rom,
+}
+
+impl Bus<u16> for Memory {
```

Comment:
If this trait is used to define a 16bits Bus you're complicating the thing too much, the gameboy has only one bus and it's always in 16bits

Reply:
The bus trait can be used to be a gate to every component recieve and send data. Registers, MBCs, Vram, they all have a channel to recieve data and send data. That channel is this Bus trait.